src/friendly-iframe-embed.js

```diff
@@ -759,85 +801,113 @@ export class FriendlyIframeEmbed {
-    installStylesLegacy(
-      childWin.document,
-      ampSharedCss,
-      /* callback */ null,
-      /* opt_isRuntimeCss */ true,
-      /* opt_ext */ 'amp-runtime'
-    );
-
-    // Run pre-install callback.
-    if (opt_preinstallCallback) {
-      opt_preinstallCallback(childWin);
-    }
-
-    // Install embeddable standard services.
-    installStandardServicesInEmbed(childWin);
-
-    // Install built-ins and legacy elements.
-    copyBuiltinElementsToChildWindow(topWin, childWin);
-    stubLegacyElements(childWin);
-
-    const promises = [];
-    extensionIds.forEach((extensionId) => {
-      // This will extend automatic upgrade of custom elements from top
-      // window to the child window.
-      if (!LEGACY_ELEMENTS.includes(extensionId)) {
-        stubElementIfNotKnown(childWin, extensionId);
-      }
+    return maybeGetDelayPromise(undefined)
+      .then(() => {
+        // Install necessary polyfills.
+        installPolyfillsInChildWindow(parentWin, childWin);
+      })
+      .then(maybeGetDelayPromise)
+      .then(() => {
+        // Install runtime styles.
+        installStylesLegacy(
+          childWin.document,
+          ampSharedCss,
+          /* callback */ null,
+          /* opt_isRuntimeCss */ true,
+          /* opt_ext */ 'amp-runtime'
+        );
+      })
+      .then(maybeGetDelayPromise)
+      .then(() => {
+        // Run pre-install callback.
+        if (opt_preinstallCallback) {
+          opt_preinstallCallback(childWin);
+        }
+      })
+      .then(maybeGetDelayPromise)
+      .then(() => {
+        // Install embeddable standard services.
+        installStandardServicesInEmbed(childWin);
+      })
+      .then(maybeGetDelayPromise)
+      .then(() => {
+        // Install built-ins and legacy elements.
+        copyBuiltinElementsToChildWindow(topWin, childWin);
+        stubLegacyElements(childWin);
+
+        extensionIds.forEach((extensionId) => {
+          // This will extend automatic upgrade of custom elements from top
+          // window to the child window.
+          if (!LEGACY_ELEMENTS.includes(extensionId)) {
+            stubElementIfNotKnown(childWin, extensionId);
+          }
+        });
+      });
+  }
 
-      // Install CSS.
-      const promise = extensions
-        .preloadExtension(extensionId)
-        .then((extension) => {
-          // Adopt embeddable extension services.
-          /** @type {!Array} */ (extension.services).forEach((service) => {
-            installServiceInEmbedIfEmbeddable(childWin, service.serviceClass);
-          });
+  /**
+   * Install extensions in the child window (friendly iframe).
+   * @param {!./service/extensions-impl.Extensions} extensions
+   * @param {!Window} childWin
+   * @param {!Array<string>} extensionIds
+   * @return {!Promise}
+   * @visibleForTesting
+   */
+  doInstallExtensionsInChildWindow(extensions, childWin, extensionIds) {
+    const parentWin = toWin(childWin.frameElement.ownerDocument.defaultView);
```

Comment:
do we need to check if this exists? Maybe the frame gets destroyed between chunks now?

Reply:
good idea. adding it.